### PR TITLE
Product

### PR DESCRIPTION
**Pull request by CasoMateo**
I made use of the suggestions EduardoSantos7 provided to create more concise, well written code. This contributes to the development of good practices and the erradication of bad ones.

### CODE REVIEW

File: Product [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class Product: 
+    def __init__(self, sale_price, number_products, purchase_price, name, brand, image, barcode, instructions, 
+        size):
+        self.sale_price = sale_price # needs to be an integer
```
Comment (EduardoSantos7) [inline Product]: Pep8 convention says that when you write an inline comment you must use 2 spaces before the comment.

File: Product [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        self.purchase_price = purchase_price # needs to be an integer
+        self.name = name # needs to be a string
+        self.brand = brand # needs to be a string 
+        self.image = image # needs to be a string
+        self.barcode = barcode # needs to be boolean 
+        self.instructions = instructions # needs to be a string 
+        self.size = size # needs to be an integer 
+
+
+    def design(): # we will need certain sizes, materials and colors
```
Comment (EduardoSantos7) [inline Product]: I don't think this is a characteristic of a product.

File: Product [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return ('We determine the size, texture and appropiate features the product should have to catch the attention of the users')
+
+        # we should return the appropiate combinations of the factors above
```
Comment (EduardoSantos7) [inline Product]: this comment is unindented or it shouldn't be here.

File: Product [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        self.instructions = instructions # needs to be a string 
+        self.size = size # needs to be an integer 
+
+
+    def design(): # we will need certain sizes, materials and colors
+        return ('We determine the size, texture and appropiate features the product should have to catch the attention of the users')
+
+        # we should return the appropiate combinations of the factors above 
+
+    def get_price_product(): # we will need to take into account factors like how difficult it was to
```
Comment (EduardoSantos7) [inline Product]: You're not using Docstrings and this is not a correct structure for python comments.

File: Product [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+        # we should return the appropiate combinations of the factors above 
+
+    def get_price_product(): # we will need to take into account factors like how difficult it was to 
+                            # create, how much it originally costs, and how much customers would be willing to 
+                            # pay for the product   
+        return ('We determine the adequate price of the product according to the cost of production and what customers would be comfortable with')
+        # we should return an adequate price, according to the factors above
+
+    def get_products_stock(): # we have to have access to the stock
```
Comment (EduardoSantos7) [inline Product]: This shouldn't be a product method.

**Review by EduardoSantos7 — CHANGES_REQUESTED** [review]
First of all, congratulations on your first PR, just a couple of comments:

 - Each PR must have a description. 
- You're not using docstrings.
- There're methods that shouldn't be in a product class. (Think in what is a real Product and what it can do.)

**Comment by EduardoSantos7** [comment]
When you add changes, especially changes about the last suggestions, you should comment or close a suggestion and add the update description in the PR body.

File: Statistics [hunk 6]
```diff
@@ -0,0 +1,104 @@
+def most_sold():
+    """We will need a list representing the number of times a product has been bought. Hence, 
+    we will need to implement a function where the user can register the sales. """
+    
+    return ('We should return the product with the hightest sales according to the table above')
+
```
Comment (EduardoSantos7) [inline Statistics:6]: (just a style/convention comment)

When you write functions on the script scope you should use 2 lines to separate functions, you can download the [PEP8 extension ](https://code.visualstudio.com/docs/python/linting)for VS Code and let the extension warn you about style errors

File: Statistics [hunk 7]
```diff
@@ -0,0 +1,104 @@
+def most_sold():
+    """We will need a list representing the number of times a product has been bought. Hence, 
+    we will need to implement a function where the user can register the sales. """
+    
+    return ('We should return the product with the hightest sales according to the table above')
+ 
+def least_sold(): 
+    """We will need a list representing the number of times a product has been bought. Hence,
```
Comment (EduardoSantos7) [inline Statistics:9]: If you need a list, why isn't a parameter here?

File: Product (hunk 5 above)
Comment (CasoMateo) [reply Product]: I removed unnecessary methods to maintain a clean and understandable framework for my program.

File: Product (hunk 4 above)
Comment (CasoMateo) [reply Product]: I modified my comments' structure, making use of docstrings for better understanding of what my program actually does.

File: Product (hunk 3 above)
Comment (CasoMateo) [reply Product]: I modified my comments' structure, making use of docstrings for better understanding of what my program actually does.

File: Product (hunk 2 above)
Comment (CasoMateo) [reply Product]: I removed unnecessary attributes to maintain a clean and understandable framework for my program.

File: Product (hunk 1 above)
Comment (CasoMateo) [reply Product]: When updating my program, I was aware of this suggestion. Hence, you can now see that I applied the Pep8 convention.

File: Statistics [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    """We should take feedback from the custumers, expressing the level of satisfaction 
+    they had when they bought the store's prdocuts""" 
+    
+    return ('We should return the feedback our customers provided to enhance or modify 
+    our services')
+
+def compare_price():
+    """We need a list other companies' products to compare the price to. """
+        
+    return ("We should return an analysis depicting our product's price compare to others")
```
Comment (EduardoSantos7) [inline Statistics:118]: Why these functions doesn't have an implementation yet?

File: StoreSupplier [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    def __init__(self, products, sell_prices, email_address, phone_number, 
+        distance_from_shop, supply_capacity):
+        self.products = products                      #  needs to be a list 
+        self.sell_prices = sell_prices                #  needs to be a list 
+        self.email_address = email_address            #  needs to be a string
+        self.phone_number = phone_number              #  needs to be an integer
+        self.distance_from_shop = distance_from_shop  #  needs to be an integer
+        self.supply_capacity = supply_capacity        #  needs to be an integer
+        
+    def s_t_store():
```
Comment (EduardoSantos7) [inline StoreSupplier]: s_t*?

Comment (CasoMateo) [reply StoreSupplier]: Sorry, it was suposed to be an abbreviation. I'll modify the code right away.

File: Statistics (hunk 6 above)
Comment (CasoMateo) [reply Statistics:6]: I downloaded the extension, and I've already modified the code with regard to your suggestion.

File: Statistics (hunk 7 above)
Comment (CasoMateo) [reply Statistics:9]: I'll put the appropiate parameters for the functions according to the information within the docstrings.

File: Product [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    def __init__(self, sale_price, number_products, purchase_price, name, brand, image, barcode, instructions, 
+        size):
+        self.sale_price = sale_price                # needs to be an integer
+        self.number_products = number_products      # needs to be an integer
+        self.purchase_price = purchase_price        # needs to be an integer
+        self.name = name                            # needs to be a string
+        self.brand = brand                          # needs to be a string 
+        self.image = image                          # needs to be a string
+        self.barcode = barcode                      # needs to be boolean 
+        self.instructions = instructions            # needs to be a string
```
Comment (EduardoSantos7) [inline Product:11]: This field could be extra?

File: Product [hunk 11]
```diff
@@ -0,0 +1,27 @@
+class Product: 
+    def __init__(self, sale_price, number_products, purchase_price, name, brand, image, barcode, instructions, 
+        size):
+        self.sale_price = sale_price                # needs to be an integer
+        self.number_products = number_products      # needs to be an integer
+        self.purchase_price = purchase_price        # needs to be an integer
```
Comment (EduardoSantos7) [inline Product:6]: This field could be extra? Is it not the function of the stock to handle that?

File: Product [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+        self.instructions = instructions            # needs to be a string 
+
+
+    def design(): 
+        """ We will need certain sizes, materials and colors. """
+        
+        return ('We determine the size, texture and appropiate features the product should have to catch the attention of the users')
+
+
+    def get_price_product():
```
Comment (EduardoSantos7) [inline Product:20]: you can avoid `product` in the signature of the method because is kind of duplication.

File: Statistics (hunk 7 above)
Comment (EduardoSantos7) [reply Statistics:9]: did you?

File: Store [hunk 13]
```diff
@@ -0,0 +1,54 @@
+class Stores:
+    def __init__(self, inventory, location, email_address, phone_number, name, number_products, cash_inflow, 
+        credit_card, debit_card, adult_products, rent, department, licenses, prduct_suggestions, employees
+        security, furniture, logo, promotions, brands):
+        self.iventory = inventory                        # needs to be a list
```
Comment (EduardoSantos7) [inline Store:5]: inventory* but you should use stock here.

**Review by EduardoSantos7 — APPROVED** [review]
Some minor changes, nothing blocking! LGTM! 👍

**Pull request closed without merging**